datasets/gb/hmt/crawler.py

```diff
@@ -123,8 +152,23 @@ def parse_row(context: Context, row: Dict[str, Any]):
     entity.add_cast("LegalEntity", "legalForm", entity_type)
 
     reg_number = row.pop("Entity_BusinessRegNumber", "")
-    entity.add_cast("LegalEntity", "registrationNumber", split_reg_no(reg_number))
-
+    reg_numbers = split_reg_no(reg_number)
+    if reg_numbers:
+        for reg_no in reg_numbers:
+            result = process_entry(reg_no, REGEX_IDENTIFIERS)
+            if result:
+                prop = context.lookup_value("properties", slugify(result["key"]))
+                if prop is not None:
+                    if prop == "kppCode":
+                        entity.add_cast("Company", prop, result["value"])
+                    else:
+                        entity.add_cast("LegalEntity", prop, result["value"])
+                else:
+                    context.log.warning(
+                        "Unknown property", value=slugify(result["key"])
+                    )
```

Comment:
I think after splitting, you can remap these props with this lookup on type.identifier https://opensanctions-corp.slack.com/archives/C07C5G9KZED/p1731064875859569 rather than adding this code

Reply:
here https://github.com/opensanctions/opensanctions/pull/1509/files#diff-573cc64c90b8256f0913c009aa2ce0d475bdf166f59e20e29a505f61f96748e4R59